### DISCUSSION TITLE
Close : Request a null or empty field in the category search.

### DISCUSSION THREAD

**Discussion by meteos77** [textbrowser/biblioteq#252] [Q&A] [answered]
I am using the book search tool and I would like BiblioteQ  to give me a list of detective story that do not have an isbn13.
(precision : detective story = RP-Roman Policier)

what is the opposite of the % symbol in the search tool

in SQL it would look like this for the % in isbn13 :
SELECT title,author,isbn13 FROM book WHERE location LIKE ('RP-Roman Policier') AND isbn13 IS NOT NULL 

but to put in the field ISBN13 : empty or none ?

in SQL it would look like this :
SELECT title,author,isbn13 FROM book WHERE location LIKE ('RP-Roman Crime') AND isbn13 IS NULL

thank you in advance

**Comment by textbrowser** [comment]
https://github.com/textbrowser/biblioteq/commit/24622790e58c908b0766bc57babd990c232257e0

**Reply by textbrowser** [reply]
This is both a question and a request.

If you have SQL questions, there are numerous online and physical resources.

https://stackoverflow.com/

BQ is free software, not free customer / technical support. :P

**Reply by textbrowser** [reply]
I do enjoy the occasional question, occasionally. I also enjoy providing assistance, occasionally.

**Reply by meteos77** [reply]
what to answer: you are the only one who can assure us of such fast evolutions

again bravo for this generosity and this availability

**Comment by textbrowser** [comment]
This is a request. Discussions cannot be closed, only deleted.

**Comment by textbrowser** [comment]
Should be a ticket.

**Reply by meteos77** [reply]
it's difficult for me to know in advance if the functionality existed with another symbol so in the discussions there is "Q&A" which was the most suitable.

If for your follow-up you want me to create a ticket no worries

**Reply by textbrowser** [reply]
If it doesn't work as expected, perhaps that's a problem.

**Comment by textbrowser** [comment, marked as answer]
https://user-images.githubusercontent.com/10701156/204310576-5f0d1c9a-4530-47a7-b7b6-6b818a23f708.mp4

**Comment by textbrowser** [comment]
> I am using the book search tool and I would like BiblioteQ to give me a list of detective story that do not have an isbn13. (precision : detective story = RP-Roman Policier)
> 
> what is the opposite of the % symbol in the search tool
> 
> in SQL it would look like this for the % in isbn13 : SELECT title,author,isbn13 FROM book WHERE location LIKE ('RP-Roman Policier') AND isbn13 IS NOT NULL
> 
> but to put in the field ISBN13 : empty or none ?
> 
> in SQL it would look like this : SELECT title,author,isbn13 FROM book WHERE location LIKE ('RP-Roman Crime') AND isbn13 IS NULL
> 
> thank you in advance

The opposite of the wildcard? A non-wildcard.  There are restrictions with null values. For example, LIKE '%' on a null value is nonsensical.

**Reply by textbrowser** [reply]
This is a reason for the existence of IS NULL.

**Comment by textbrowser** [comment]
Not a simple discussion.

https://www.sqlite.org/nulls.html

**Reply by meteos77** [reply]
I'm not there yet with SQL but it's good to know for future queries thanks for the link

**Comment by textbrowser** [comment]
https://www.sqlite.org/lang_corefunc.html#coalesce

**Comment by textbrowser** [comment]
The recent change will address null ISBNs and non-null ISBNs. For null ISBNs, it's expected that both ISBN-10 and ISBN-13 are null because and / or operators. If you have an ISBN, it can be converted into another form so you shouldn't have one but not the other. If you do, that's a fault on the administrator. I don't have a trigger to enforce that both are present if one is. Anyway.

**Reply by meteos77** [reply]
In our main database we have no isbn-10.  Do you advise to fill in isbn10 if isbn13 present?

**Comment by meteos77** [comment]
currently my tests 
search with ISBN-13 left as is (so displays ISBN-13)
result: NULLS

search with ISBN-13 with %
result: NULLs and FILLED

search with ISBN-13 with 9%
result: the FILLED
